shipment_advice_planner/wizards/shipment_advice_planner.py

```diff
@@ -0,0 +1,151 @@
+
+    picking_to_plan_ids = fields.Many2many(
+        comodel_name="stock.picking",
+        string="Pickings to plan",
+        required=True,
+        domain='[("can_be_planned_in_shipment_advice", "=", True),'
+        '("warehouse_id", "=?", warehouse_id),]',
+        compute="_compute_picking_to_plan_ids",
+        store=True,
+        readonly=False,
+    )
+    shipment_planning_method = fields.Selection(
+        selection=[("simple", "Simple")], required=True, default="simple"
+    )
+    warehouse_id = fields.Many2one(comodel_name="stock.warehouse")
+    dock_id = fields.Many2one(
+        comodel_name="stock.dock", domain='[("warehouse_id", "=", warehouse_id)]'
+    )
+
+    @api.constrains("warehouse_id", "dock_id", "picking_to_plan_ids")
+    def _check_warehouse(self):
+        for rec in self:
+            if not rec.warehouse_id:
+                continue
+            if rec.dock_id and rec.dock_id.warehouse_id != rec.warehouse_id:
+                raise ValidationError(
+                    _("The dock doesn't belong to the selected warehouse.")
+                )
+            if (
+                rec.picking_to_plan_ids
+                and rec.picking_to_plan_ids.warehouse_id != rec.warehouse_id
+            ):
+                raise ValidationError(
+                    _("The transfers don't belong to the selected warehouse.")
+                )
+
+    @api.onchange("warehouse_id", "dock_id", "picking_to_plan_ids")
+    def _onchange_check_warehouse(self):
+        self.ensure_one()
+        self._check_warehouse()
+
+    @api.depends("picking_to_plan_ids")
+    def _check_picking_to_plan(self):
+        for rec in self:
+            if rec.picking_to_plan_ids.filtered(
+                lambda p: not p.can_be_planned_in_shipment_advice
+            ):
+                raise ValidationError(
+                    _("The transfers selected must be ready and of the delivery type.")
+                )
+
+    @api.onchange("picking_to_plan_ids")
+    def _onchange_check_picking_to_plan(self):
+        self.ensure_one()
+        self._check_picking_to_plan()
+
+    @api.model
+    def _get_compute_picking_to_plan_ids_depends(self):
+        return ["shipment_planning_method", "warehouse_id"]
+
+    @api.depends(lambda m: m._get_compute_picking_to_plan_ids_depends())
+    def _compute_picking_to_plan_ids(self):
+        """meant to be inherited if for specific method picking to plan should have
+        a default record set"""
+        active_model = self.env.context.get("active_model")
+        active_ids = self.env.context.get("active_ids")
+        if not active_ids or active_model != "stock.picking":
+            return self.update({"picking_to_plan_ids": False})
+        pickings_to_plan = self.env[active_model].search(
+            [
+                ("id", "in", active_ids),
+                ("can_be_planned_in_shipment_advice", "=", True),
+                ("warehouse_id", "=?", self.warehouse_id.id),
```

Comment:
The warehouse is False when the wizard is launched, isn't it?

Reply:
yes